src/connection.c

```diff
@@ -0,0 +1,451 @@
+                locality_city =
+                    pn_ssl_get_remote_subject_subfield(pn_ssl(transport), PN_SSL_CERT_SUBJECT_CITY_OR_LOCALITY);
+
+                if (locality_city) {
+                    uid_length += strlen((const char*) locality_city);
+                    semi_colon_count++;
+                }
+            } else if (components[x] == CERT_ORGANIZATION_NAME) {
+                organization =
+                    pn_ssl_get_remote_subject_subfield(pn_ssl(transport), PN_SSL_CERT_SUBJECT_ORGANIZATION_NAME);
+
+                if (organization) {
+                    uid_length += strlen((const char*) organization);
+                    semi_colon_count++;
+                }
+            } else if (components[x] == CERT_ORGANIZATION_UNIT) {
+                org_unit = pn_ssl_get_remote_subject_subfield(pn_ssl(transport), PN_SSL_CERT_SUBJECT_ORGANIZATION_UNIT);
+
+                if (org_unit) {
+                    uid_length += strlen((const char*) org_unit);
+                    semi_colon_count++;
+                }
+            } else if (components[x] == CERT_COMMON_NAME) {
+                common_name = pn_ssl_get_remote_subject_subfield(pn_ssl(transport), PN_SSL_CERT_SUBJECT_COMMON_NAME);
+
+                if (common_name) {
+                    uid_length += strlen((const char*) common_name);
+                    semi_colon_count++;
+                }
+            } else if (components[x] == CERT_FINGERPRINT_SHA1 || components[x] == CERT_FINGERPRINT_SHA256 ||
+                       components[x] == CERT_FINGERPRINT_SHA512) {
+                // Allocate the memory for message digest
+                int out                = 0;
+                int fingerprint_length = 0;
+
+                if (components[x] == CERT_FINGERPRINT_SHA1) {
+                    fingerprint_length = 40;
+                    out = pn_ssl_get_cert_fingerprint(pn_ssl(transport), fingerprint, fingerprint_length + 1,
+                                                      PN_SSL_SHA1);
+                } else if (components[x] == CERT_FINGERPRINT_SHA256) {
+                    fingerprint_length = 64;
+                    out = pn_ssl_get_cert_fingerprint(pn_ssl(transport), fingerprint, fingerprint_length + 1,
+                                                      PN_SSL_SHA256);
+                } else if (components[x] == CERT_FINGERPRINT_SHA512) {
+                    fingerprint_length = 128;
+                    out = pn_ssl_get_cert_fingerprint(pn_ssl(transport), fingerprint, fingerprint_length + 1,
+                                                      PN_SSL_SHA512);
+                }
+
+                // Avoid 'out unused' compiler warnings if NDEBUG undef'ed
+                (void) out;
+
+                assert(out != PN_ERR);
+
+                uid_length += fingerprint_length;
+                semi_colon_count++;
+            } else {
+                // This is an unrecognized component.  Log a critical
+                // error.
```

Comment:
clang-format did not reflow this one. It will add breaks if a comment is over column limit, but it apparently does not try to join lines. Makes sense, joining lines can easily break meaningful formatting.